go-controller/pkg/libovsdb/libovsdb.go

```diff
@@ -31,7 +36,40 @@ import (
 func newClient(cfg config.OvnAuthConfig, dbModel model.ClientDBModel, stopCh <-chan struct{}, opts ...client.Option) (client.Client, error) {
 	const connectTimeout time.Duration = types.OVSDBTimeout * 2
 	const inactivityTimeout time.Duration = types.OVSDBTimeout * 18
-	logger := klogr.New()
+
+	var logger logr.Logger
+
+	logggerFilename := config.Logging.LibovsdbFile
+	if len(logggerFilename) == 0 {
+		// Not using a separate log file for libovsdb client
+		logger = klogr.New()
+	} else {
+		// Sanity check: Make sure logger file can be opened
```

Comment:
I was thinking that this diverts all libovsdb logs to a separate file and among those there might be important events we would still like to log in the main logs files (reconnections, cache/async errors, etc...)

klog allows to set output by severity but I don't think that's flexible enought for us

To avoid this I see two options:

- Support to log transactions to a separate log file natively in libovsdb client
- Maybe implement a `logr.LogSink` that delegates to klog/stgr logger sinks depending on level

Reply:
> hmm.... not all libovsdb logs, @jcaamano it will only divert logs from the client. Any particular log (file/line) in libovsdb you are concerned about? Another approach would be to provide the libovsdb client with 2 loggers (i.e. add WithTransactionLogger(logger2)) in the libovsdb codebase. Wdyt?

I guess I would be interested in the internal reconnect logs, or any errors the client has processing updates from the server.

https://github.com/ovn-org/ovn-kubernetes/blob/fa9028b9ffd07d0fed076f2bba5352ced75e4438/go-controller/vendor/github.com/ovn-org/libovsdb/client/client.go#L1296
https://github.com/ovn-org/ovn-kubernetes/blob/fa9028b9ffd07d0fed076f2bba5352ced75e4438/go-controller/vendor/github.com/ovn-org/libovsdb/cache/cache.go#L935

But the approach could probably be different ans instead select which logs should go to that other transaction specific logger

WithTransactionLogger I think makes sense

Reply:
Resolving, let's defer this to later.